src/tools/perf/lib/libperf.c

```diff
@@ -1267,6 +1282,163 @@ static ucs_status_t ucp_perf_test_send_local_data(ucx_perf_context_t *perf,
     return status;
 }
 
+static void err_cb(void *arg, ucp_ep_h ep, ucs_status_t status)
+{
+    ucs_diag("error handling callback was invoked with status %d (%s)", status,
+             ucs_status_string(status));
+}
+
+/*
+ * Pack array length and array contents into a buffer.
+ *
+ * @param [out] ptr    Pointer to the buffer.
+ * @param [in]  length Length of the array.
+ * @param [in]  value  Pointer to the array.
+ */
+static void ucp_perf_pack_array(void **ptr, uint64_t length, const void *value)
+{
+    *ucs_serialize_next(ptr, uint64_t) = length;
+    if (length > 0) {
```

Comment:
why needed ? to avoid static check/coverity error?

Reply:
won't the next line `memcpy(ucs_serialize_next_raw(ptr, void, length), value, length);` be no-operation anyway when length==0?

Reply:
Yes, it will be no-op
Personally I would still prefer to have that explicit check before the function call to make the intention clear. But if you have strong opinion on that - I can remove that condition, not a big deal anyway

Reply:
Static check complains on this change. So I put the condition back
```
static checkers found errors:
Error: CLANG_WARNING:
/src/tools/perf/lib/libperf.c:1326:5: warning: Null pointer passed to 2nd parameter expecting 'nonnull' [core.NonNullParamChecker]
#    memcpy(ucs_serialize_next_raw(ptr, void, length), value, length);
```